akka-persistence-testkit/src/main/scala/akka/persistence/testkit/PersistenceTestKitImpl.scala

```diff
@@ -0,0 +1,237 @@
+package akka.persistence.testkit
+
+import java.util.UUID
+import java.util.concurrent.ConcurrentHashMap
+import java.util.function.BiFunction
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.persistence.{ AtomicWrite, PersistentRepr }
+import akka.testkit.TestKitBase
+import com.typesafe.config.{ Config, ConfigFactory }
+import akka.persistence.journal.AsyncWriteJournal
+
+import scala.concurrent.Future
+import scala.collection.immutable
+import scala.util.{ Failure, Success, Try }
+
+trait PersistenceTestKit extends TestKitBase with PersistentTestKitOps {
+
+  override implicit lazy val system = {
+    //todo probably implement method for setting plugin in Persistence for testing purposes
+    ActorSystem(
+      s"persistence-testkit-${UUID.randomUUID()}",
+      PersistenceTestKitPlugin.PersitenceTestkitPluginConfig
+        .withFallback(ConfigFactory.defaultApplication()))
```

Comment:
`defaultApplication` should not be mandatory, it must be passed in from concrete test.

However, might be better to not extend TestKit, and instead provide this as a utility than something that is mixed in to the test?

Reply:
that would also make it safer to add more things later, without risking name clash with user's test